tools/bug-report/pkg/kubectlcmd/kubectlcmd.go

```diff
@@ -0,0 +1,130 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubectlcmd
+
+import (
+	"bytes"
+	"fmt"
+	"os/exec"
+	"strings"
+	"time"
+
+	"istio.io/istio/operator/pkg/util"
+	"istio.io/pkg/log"
+)
+
+// Options contains the Run options.
+type Options struct {
+	// Path to the kubeconfig file.
+	Kubeconfig string
+	// ComponentName of the kubeconfig context to use.
+	Context string
+
+	// namespace - k8s namespace for Run command
+	Namespace string
+
+	// DryRun performs all steps but only logs the Run command without running it.
+	DryRun bool
+	// Maximum amount of time to wait for resources to be ready after install when Wait=true.
+	WaitTimeout time.Duration
+
+	// output - output mode for Run i.e. --output.
+	Output string
+
+	// extraArgs - more args to be added to the Run command, which are appended to
+	// the end of the Run command.
+	ExtraArgs []string
+}
+
+// Logs returns the logs for the given namespace/pod/container.
+func Logs(namespace, pod, container string, previous, dryRun bool) (string, error) {
+	cmdStr := []string{"logs"}
+	if previous {
+		cmdStr = append(cmdStr, "-p")
+	}
+	stdout, err := Run(cmdStr,
+		&Options{
+			Namespace: namespace,
+			ExtraArgs: []string{pod, "-c", container},
+			DryRun:    dryRun,
+		})
+	if err != nil {
+		return "", err
+	}
+	return stdout, nil
+}
+
+// Exec runs exec for the given command in the given namespace/pod/container.
+func Exec(namespace, pod, container string, dryRun bool, command ...string) (string, error) {
```

Comment:
Why not use client-go?

Reply:
it will end up being way more verbose and i think will discourage people from adding commands to fetch new artifacts. i get that os.run is not generally a great option, but i wanted to retain some of the simplicity of running kubectl commands - so it's somewhat clear at least what command is being run and easy to add or tweak.

Reply:
I think its a good idea. Its only more verbose since you are not use pkg/kube, from what I can tell. making this shell out means we need to deal with all the junk from relying on the right kubectl being in the path, etc